### PR TITLE
Package azure-cosmos-db.0.2.0

### PR DESCRIPTION
**Pull request by mknaack**
### `azure-cosmos-db.0.2.0`
Azure cosmos db interface
Provides an interface to the Azure Cosmos db.

---
* Homepage: https://bitbucket.org/knaack/azure-cosmos-db
* Source repo: git+https://bitbucket.org/knaack/azure-cosmos-db.git
* Bug tracker: https://bitbucket.org/knaack/azure-cosmos-db/issues

---
:camel: Pull-request generated by opam-publish v2.1.0

### CODE REVIEW

File: packages/azure-cosmos-db/azure-cosmos-db.0.2.0/opam [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  "ssl"
+  "atdgen"
+  "base64" {>= "3.5.0"}
+  "uri"
+  "cohttp"
+  "cohttp-lwt-unix"
+  "cryptokit"
+  "alcotest-lwt" {with-test}
+]
+homepage: "https://bitbucket.org/knaack/azure-cosmos-db"
```
Comment (mseri) [inline packages/azure-cosmos-db/azure-cosmos-db.0.2.0/opam:24]:
```suggestion
conflicts: [ "result" {< "1.5"} ]
homepage: "https://bitbucket.org/knaack/azure-cosmos-db"
```

Comment (mseri) [reply packages/azure-cosmos-db/azure-cosmos-db.0.2.0/opam:24]: Due to the lower bound failure
```
#       ocamlc src/.cosmos.objs/byte/cosmos__Databases.{cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlc.opt -w -40 -g -bin-annot -I src/.cosmos.objs/byte -I /home/opam/.opam/4.08/lib/astring -I /home/opam/.opam/4.08/lib/atd -I /home/opam/.opam/4.08/lib/atdgen -I /home/opam/.opam/4.08/lib/base -I /home/opam/.opam/4.08/lib/base/caml -I /home/opam/.opam/4.08/lib/base/shadow_stdlib -I /home/opam/.opam/4.08/lib/base64 -I /home/opam/.opam/4.08/lib/biniou -I /home/opam/.opam/4.08/lib/bytes -I /home/opam/.opam/4.08/lib/cohttp -I /home/opam/.opam/4.08/lib/cohttp-lwt -I /home/opam/.opam/4.08/lib/cohttp-lwt-unix -I /home/opam/.opam/4.08/lib/conduit -I /home/opam/.opam/4.08/lib/conduit-lwt -I /home/opam/.opam/4.08/lib/conduit-lwt-unix -I /home/opam/.opam/4.08/lib/cryptokit -I /home/opam/.opam/4.08/lib/easy-format -I /home/opam/.opam/4.08/lib/fieldslib -I /home/opam/.opam/4.08/lib/fmt -I /home/opam/.opam/4.08/lib/ipaddr -I /home/opam/.opam/4.08/lib/ipaddr/unix -I /home/opam/.opam/4.08/lib/logs -I /home/opam/.opam/4.08/lib/lwt -I /home/opam/.opam/4.08/lib/lwt/unix -I /home/opam/.opam/4.08/lib/lwt_ssl -I /home/opam/.opam/4.08/lib/magic-mime -I /home/opam/.opam/4.08/lib/mmap -I /home/opam/.opam/4.08/lib/ocaml/threads -I /home/opam/.opam/4.08/lib/parsexp -I /home/opam/.opam/4.08/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.08/lib/re -I /home/opam/.opam/4.08/lib/re/posix -I /home/opam/.opam/4.08/lib/result -I /home/opam/.opam/4.08/lib/seq -I /home/opam/.opam/4.08/lib/sexplib -I /home/opam/.opam/4.08/lib/sexplib0 -I /home/opam/.opam/4.08/lib/ssl -I /home/opam/.opam/4.08/lib/stringext -I /home/opam/.opam/4.08/lib/uchar -I /home/opam/.opam/4.08/lib/uri -I /home/opam/.opam/4.08/lib/uri/services -I /home/opam/.opam/4.08/lib/uri/sexp -I /home/opam/.opam/4.08/lib/yojson -I /home/opam/.opam/4.08/lib/zarith -intf-suffix .ml -no-alias-deps -open Cosmos -o src/.cosmos.objs/byte/cosmos__Databases.cmo -c -impl src/databases.pp.ml)
# File "src/databases.ml", line 176, characters 33-42:
# 176 |     if code = expected_code then Result.ok expected_code
#                                        ^^^^^^^^^
# Error: Unbound value Result.ok
#     ocamlopt src/.cosmos.objs/native/cosmos__Databases.{cmx,o} (exit 2)
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlopt.opt -w -40 -g -I src/.cosmos.objs/byte -I src/.cosmos.objs/native -I /home/opam/.opam/4.08/lib/astring -I /home/opam/.opam/4.08/lib/atd -I /home/opam/.opam/4.08/lib/atdgen -I /home/opam/.opam/4.08/lib/base -I /home/opam/.opam/4.08/lib/base/caml -I /home/opam/.opam/4.08/lib/base/shadow_stdlib -I /home/opam/.opam/4.08/lib/base64 -I /home/opam/.opam/4.08/lib/biniou -I /home/opam/.opam/4.08/lib/bytes -I /home/opam/.opam/4.08/lib/cohttp -I /home/opam/.opam/4.08/lib/cohttp-lwt -I /home/opam/.opam/4.08/lib/cohttp-lwt-unix -I /home/opam/.opam/4.08/lib/conduit -I /home/opam/.opam/4.08/lib/conduit-lwt -I /home/opam/.opam/4.08/lib/conduit-lwt-unix -I /home/opam/.opam/4.08/lib/cryptokit -I /home/opam/.opam/4.08/lib/easy-format -I /home/opam/.opam/4.08/lib/fieldslib -I /home/opam/.opam/4.08/lib/fmt -I /home/opam/.opam/4.08/lib/ipaddr -I /home/opam/.opam/4.08/lib/ipaddr/unix -I /home/opam/.opam/4.08/lib/logs -I /home/opam/.opam/4.08/lib/lwt -I /home/opam/.opam/4.08/lib/lwt/unix -I /home/opam/.opam/4.08/lib/lwt_ssl -I /home/opam/.opam/4.08/lib/magic-mime -I /home/opam/.opam/4.08/lib/mmap -I /home/opam/.opam/4.08/lib/ocaml/threads -I /home/opam/.opam/4.08/lib/parsexp -I /home/opam/.opam/4.08/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.08/lib/re -I /home/opam/.opam/4.08/lib/re/posix -I /home/opam/.opam/4.08/lib/result -I /home/opam/.opam/4.08/lib/seq -I /home/opam/.opam/4.08/lib/sexplib -I /home/opam/.opam/4.08/lib/sexplib0 -I /home/opam/.opam/4.08/lib/ssl -I /home/opam/.opam/4.08/lib/stringext -I /home/opam/.opam/4.08/lib/uchar -I /home/opam/.opam/4.08/lib/uri -I /home/opam/.opam/4.08/lib/uri/services -I /home/opam/.opam/4.08/lib/uri/sexp -I /home/opam/.opam/4.08/lib/yojson -I /home/opam/.opam/4.08/lib/zarith -intf-suffix .ml -no-alias-deps -open Cosmos -o src/.cosmos.objs/native/cosmos__Databases.cmx -c -impl src/databases.pp.ml)
# File "src/databases.ml", line 176, characters 33-42:
# 176 |     if code = expected_code then Result.ok expected_code
#                                        ^^^^^^^^^
# Error: Unbound value Result.ok
```

**Comment by mseri** [comment]
The ocaml 5 failure is due to incompatibility of atdgen.1.9.1, I will look at it separately

**Comment by mseri** [comment]
The tests are incompatible with 32 bits architectures. Is this just the tests or the library should not be used on 32 bits?
```

#=== ERROR while compiling azure-cosmos-db.0.2.0 ==============================#
# context              2.2.0~alpha~dev | linux/arm32 | ocaml-base-compiler.4.14.0 | pinned(https://bitbucket.org/knaack/azure-cosmos-db/get/b3af93ec331f8c3dee77d88b9576cf4234187c90.tar.gz)
# path                 ~/.opam/4.14/.opam-switch/build/azure-cosmos-db.0.2.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune runtest -p azure-cosmos-db -j 159
# exit-code            1
# env-file             ~/.opam/log/azure-cosmos-db-7-10b799.env
# output-file          ~/.opam/log/azure-cosmos-db-7-10b799.out
### output ###
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlc.opt -w -40 -annot -short-paths -g -bin-annot -I test/.test.eobjs/byte -I /home/opam/.opam/4.14/lib/alcotest -I /home/opam/.opam/4.14/lib/alcotest-lwt -I /home/opam/.opam/4.14/lib/alcotest/engine -I /home/opam/.opam/4.14/lib/alcotest/stdlib_ext -I /home/opam/.opam/4.14/lib/angstrom -I /home/opam/.opam/4.14/lib/astring -I /home/opam/.opam/4.14/lib/atdgen -I /home/opam/.opam/4.14/lib/atdgen-runtime -I /home/opam/.opam/4.14/lib/base/caml -I /home/opam/.opam/4.14/lib/base64 -I /home/opam/.opam/4.14/lib/bigstringaf -I /home/opam/.opam/4.14/lib/biniou -I /home/opam/.opam/4.14/lib/camlp-streams -I /home/opam/.opam/4.14/lib/cmdliner -I /home/opam/.opam/4.14/lib/cohttp -I /home/opam/.opam/4.14/lib/cohttp-lwt -I /home/opam/.opam/4.14/lib/cohttp-lwt-unix -I /home/opam/.opam/4.14/lib/conduit -I /home/opam/.opam/4.14/lib/conduit-lwt -I /home/opam/.opam/4.14/lib/conduit-lwt-unix -I /home/opam/.opam/4.14/lib/cryptokit -I /home/opam/.opam/4.14/lib/domain-name -I /home/opam/.opam/4.14/lib/easy-format -I /home/opam/.opam/4.14/lib/fmt -I /home/opam/.opam/4.14/lib/ipaddr -I /home/opam/.opam/4.14/lib/ipaddr-sexp -I /home/opam/.opam/4.14/lib/ipaddr/unix -I /home/opam/.opam/4.14/lib/logs -I /home/opam/.opam/4.14/lib/lwt -I /home/opam/.opam/4.14/lib/lwt/unix -I /home/opam/.opam/4.14/lib/lwt_ssl -I /home/opam/.opam/4.14/lib/macaddr -I /home/opam/.opam/4.14/lib/magic-mime -I /home/opam/.opam/4.14/lib/ocaml/threads -I /home/opam/.opam/4.14/lib/ocplib-endian -I /home/opam/.opam/4.14/lib/ocplib-endian/bigstring -I /home/opam/.opam/4.14/lib/parsexp -I /home/opam/.opam/4.14/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.14/lib/re -I /home/opam/.opam/4.14/lib/seq -I /home/opam/.opam/4.14/lib/sexplib -I /home/opam/.opam/4.14/lib/sexplib0 -I /home/opam/.opam/4.14/lib/ssl -I /home/opam/.opam/4.14/lib/stdlib-shims -I /home/opam/.opam/4.14/lib/stringext -I /home/opam/.opam/4.14/lib/uri -I /home/opam/.opam/4.14/lib/uri-sexp -I /home/opam/.opam/4.14/lib/uri/services -I /home/opam/.opam/4.14/lib/uutf -I /home/opam/.opam/4.14/lib/yojson -I /home/opam/.opam/4.14/lib/zarith -I src/.cosmos.objs/byte -no-alias-deps -open Dune__exe -o test/.test.eobjs/byte/dune__exe__Test_utility.cmo -c -impl test/test_utility.pp.ml)
# File "test/test_utility.ml", line 68, characters 13-23:
# 68 |        _ts = 1526743684;
#                   ^^^^^^^^^^
# Error: Integer literal exceeds the range of representable integers of type int
```

File: packages/azure-cosmos-db/azure-cosmos-db.0.2.0/opam (hunk 1 above)
Comment (mknaack) [reply packages/azure-cosmos-db/azure-cosmos-db.0.2.0/opam:25]: Thanks for looking into this. I have updated the PR.

**Comment by mseri** [comment]
One last thing. The NNP (no naked pointers) check found the following:
```
#=== ERROR while compiling azure-cosmos-db.0.2.0 ==============================#
# context              2.2.0~alpha~dev | linux/x86_64 | ocaml-options-only-nnpchecker.1 | pinned(https://bitbucket.org/knaack/azure-cosmos-db/get/e7efe7b9564ab13e5aad5646f34e05d814456c1b.tar.gz)
# path                 ~/.opam/4.14/.opam-switch/build/azure-cosmos-db.0.2.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune runtest -p azure-cosmos-db -j 31
# exit-code            1
# env-file             ~/.opam/log/azure-cosmos-db-7-139b6f.env
# output-file          ~/.opam/log/azure-cosmos-db-7-139b6f.out
### output ###
# File "test/dune", line 2, characters 7-11:
# 2 |  (name test)
#            ^^^^
# (cd _build/default/test && ./test.exe)
# Testing `Main tests'.
# This run has ID `L4YCLNR1'.
# 
#   [OK]          utility test                0   x_ms_date.
#   [OK]          utility test                1   adjust_host_no_change.
#   [OK]          utility test                2   adjust_host_change.
#   [OK]          utility test                3   auth_key_test.
#   [OK]          utility test                4   auth_key_test_get_list.
#   [OK]          utility test                5   convert_list_databases_test.
#   [OK]          utility test                6   string_of_bool_true.
#   [OK]          utility test                7   string_of_bool_false.
# 
# Full test results in `~/.opam/4.14/.opam-switch/build/azure-cosmos-db.0.2.0/_build/default/test/_build/_tests/Main tests'.
# Test Successful in 0.002s. 8 tests run.
# Out-of-heap pointer at 0x559671152338 of value 0x55967100b320 has non-black head (tag=65)
# 
# Out-of-heap pointers were detected by the runtime.
# The process would otherwise have terminated normally.
```
This is not necessarily a blocker. Let me know if you intend to address it before I merge

**Comment by kit-ty-kate** [comment]
> One last thing. The NNP (no naked pointers) check found the following:

azure-cosmos-db does not have any C stubs. It looks like the problem comes from `cryptokit` whose bindings are not OCaml 5 / NNP ready. I'll send a PR to fix the cryptokit package and ping upstream about it.

**Comment by mseri** [comment]
Then I am going to merge

**Comment by mseri** [comment]
Thanks a lot

**Pull request merged**